datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRolesController.java

```diff
@@ -0,0 +1,307 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.datavaultplatform.webapp.model.RoleViewModel;
+
+import java.util.*;
+//import java.util.stream.Collectors;
+
+
+@Controller
+public class AdminRolesController {
+
+    private static final Logger logger = LoggerFactory.getLogger(AdminRolesController.class);
+    private RestService restService;
+
+    public void setRestService(RestService restService) {
+        this.restService = restService;
+    }
+
+    private void setCommonModel(ModelMap model) {
+
+
+        List<RoleModel> roles = restService.getEditableRoles();
+
+        if (roles.isEmpty()) {
+            logger.info("getRolesListing: roles has no items in rest fetch");
+        }
+
+        model.addAttribute("roles", roles);
+
+
+        if (!model.containsAttribute("errormessage")) {
+            model.addAttribute("errormessage", "");
+        }
+
+
+    }
+
+
+    @RequestMapping(value = "/admin/roles", method = RequestMethod.GET)
+    public String getRolesListing(ModelMap model) {
+
+        try {
+                setCommonModel(model);
+
+        } catch (Exception e) {
+
+            logger.error("Exception in getRolesListing ", e);
+            model.addAttribute("errormessage", "<strong>Error loading roles:</strong>" + e.getMessage());
+
+        } finally {
+            return "admin/roles/index";
+        }
+    }
+
+    @RequestMapping(value = "/admin/roles/create", method = RequestMethod.POST)
+    public String addRole(@ModelAttribute RoleModel role, ModelMap model, @RequestParam String action) throws Exception {
+
+        setCommonModel(model);
+        return "admin/roles/index";
+    }
+
+    private RoleModel getRoleById(long roleid) {
+
+        Optional<RoleModel> optional = restService.getEditableRoles().stream().filter(x -> x.getId().equals(roleid)).findFirst();
+        RoleModel role = optional.isPresent() ? optional.get() : null;
+
+        return role;
+
+    }
+
+
+    @RequestMapping(value = "/admin/roles/save", method = RequestMethod.POST)
+    @ResponseBody
+    public void save(@RequestParam(value="id") int id,
+                     @RequestParam(value="name") String name,
+                     @RequestParam(value="type") String type,
+                     @RequestParam(value="description",required=false) String description,
+                     @RequestParam(value="permissions[]", required=false) String[] permissions) throws Exception {
+
+        try {
+
+            for (String p: permissions ) {
```

Comment:
There's a bit too much logging going on here. We should reduce all the info() calls into a single call.